homeassistant/components/ambient_network/entity.py

```diff
@@ -0,0 +1,49 @@
+"""Base entity class for the Ambient Weather Network integration."""
+from __future__ import annotations
+
+from abc import abstractmethod
+
+from homeassistant.core import callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import AmbientNetworkDataUpdateCoordinator
+
+
+class AmbientNetworkEntity(CoordinatorEntity[AmbientNetworkDataUpdateCoordinator]):
+    """Entity class for Ambient network devices."""
+
+    _attr_attribution = "Data provided by ambientnetwork.net"
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: AmbientNetworkDataUpdateCoordinator,
+        description: EntityDescription,
+        mac_address: str,
+    ) -> None:
+        """Initialize the Ambient network entity."""
+
+        super().__init__(coordinator)
+        self.entity_description = description
+        self._device_id = mac_address
+        self._attr_unique_id = f"{mac_address}_{description.key}"
+        self._attr_device_info = DeviceInfo(
```

Comment:
Shouldn't this integration be considered a service integration? If yes, we should set entry type:
```suggestion
        self._attr_device_info = DeviceInfo(
            entry_type=DeviceEntryType.SERVICE,
```

Reply:
Fixed.